pkg/controllers/machine_controller.go

```diff
@@ -360,11 +366,12 @@ func (r *MachineReconciler) deleteMachine(ctx context.Context, log logr.Logger,
 	}
 
 	if machine.Spec.ShutdownAt.IsZero() {
+		machine.Status.State = api.MachineStateTerminating
 		machine.Spec.ShutdownAt = time.Now()
 		if _, err := r.machines.Update(ctx, machine); err != nil {
-			return false, fmt.Errorf("failed to update ShutdownAt: %w", err)
+			return false, fmt.Errorf("failed to update ShutdownAt and State: %w", err)
 		}
-		log.V(1).Info("Updated machine ShutdownAt", "ShutdownAt", machine.Spec.ShutdownAt)
+		log.V(1).Info("Updated machine state", "ShutdownAt", machine.Spec.ShutdownAt, "State", machine.Status.State)
```

Comment:
Can we replace `state` with `object` here, as ShutdownAt is part of the spec, and state is a part of the status.

Reply:
What do you think about: Updated API machine?
I think we have similar strings somewhere else in code.

Reply:
That sounds good to me.

Reply:
fixed